lib/auth/keystore/manager.go

```diff
@@ -456,6 +516,33 @@ func (m *Manager) GetJWTSigner(ctx context.Context, ca types.CertAuthority) (cry
 	return nil, trace.NotFound("no usable JWT key pairs found")
 }
 
+// GetDecrypter returns the [crypto.Decrypter] associated with a given EncryptionKeyPair if accessible.
+func (m *Manager) GetDecrypter(ctx context.Context, keyPair *types.EncryptionKeyPair) (crypto.Decrypter, error) {
```

Comment:
in reality if you have multiple auths with PKCS#11 HSMs, you're going to have multiple possible EncryptionKeyPairs, and I don't think you're going to know which one to pass here. This is why the signer equivalents like GetJWTSigner pass the whole CA and then the keystore.Manager selects the key it can actually use with the available keystore backends. but maybe it will be better to just refactor this in a later PR

Reply:
I account for this in a later PR by calling `GetDecrypter` for each EncryptionKeyPair. I could modify this to accept a slice or an iterator of EncryptionkeyPairs, but it actually ends up being useful to know the result for each key during a rotation as a check for which keys are accessible to a given `keystore.Manager`. In particular when a key is being rotated the `Manager` can resolve both the rotated pair and the new pair, but the caller needs to make decisions based on the state of both of them (e.g. if the new pair is still unfulfilled, the key marked as rotating is still the active key. If it's fulfilled then the rotating key needs to be marked as rotated).